src/main/java/com/pinterest/secor/reader/MessageReader.java

```diff
@@ -65,7 +66,12 @@ public MessageReader(SecorConfig config, OffsetTracker offsetTracker) throws
 
         mConsumerConnector = Consumer.createJavaConsumerConnector(createConsumerConfig());
 
-        TopicFilter topicFilter = new Whitelist(mConfig.getKafkaTopicFilter());
+        if (!mConfig.getKafkaTopicBlacklist().isEmpty() && !mConfig.getKafkaTopicFilter().isEmpty()) {
+            throw new RuntimeException("Topic filter and blacklist cannot be both specified.");
+        }
+        TopicFilter topicFilter = !mConfig.getKafkaTopicBlacklist().isEmpty()? new Blacklist(mConfig.getKafkaTopicBlacklist()):
```

Comment:
If both empty, it will behavior same as current code return a WhileList(""). Not sure if it matches everything or nothing.